engines_test.go

```diff
@@ -8,15 +8,27 @@ import (
 	"testing"
 
 	. "github.com/sashabaranov/go-openai"
+	"github.com/sashabaranov/go-openai/internal/test"
 	"github.com/sashabaranov/go-openai/internal/test/checks"
 )
 
+// Helper function to test the ListEngines endpoint.
+func RandomEngine() Engine {
+	return Engine{
+		ID:     test.RandomString(),
+		Object: test.RandomString(),
+		Owner:  test.RandomString(),
+		Ready:  test.RandomBool(),
+	}
+}
+
 // TestGetEngine Tests the retrieve engine endpoint of the API using the mocked server.
 func TestGetEngine(t *testing.T) {
 	client, server, teardown := setupOpenAITestServer()
 	defer teardown()
 	server.RegisterHandler("/v1/engines/text-davinci-003", func(w http.ResponseWriter, r *http.Request) {
-		resBytes, _ := json.Marshal(Engine{})
+		engine := RandomEngine()
```

Comment:
By keeping `engine` variable inside the handler we have no way to compare it to the client-returned result. Also, the comparison would need to be pretty long as we would need to compare all four fields (one would probably be enough)

Reply:
I'll move `engine := RandomEngine()` outside of the handler and add some comparison logic that compares only one field.